File: src/main/java/com/swiggy/swiggy_backend/service/MenuService.java
package com.swiggy.swiggy_backend.service;

import java.util.List;

import com.swiggy.swiggy_backend.dto.MenuRequest;
import com.swiggy.swiggy_backend.dto.MenuResponse;

public interface MenuService {

    MenuResponse addMenuItem(Long restaurantId, MenuRequest request);

    List<MenuResponse> getMenuByRestaurant(Long restaurantId);

    MenuResponse getMenuItemById(Long id);

    MenuResponse updateMenuItem(Long id, MenuRequest request);

    void deleteMenuItem(Long id);

    List<MenuResponse> searchMenu(String keyword);
    
    void uploadMenuItemImage(Long menuItemId, String imageUrl);
}

File: swiggy-frontend/src/services/menuService.js
import api from "./api";


export const getMenuByRestaurant = (restaurantId) => {
    return api.get(
        `/restaurants/${restaurantId}/menu`
    );
};


export const getMenuItemById = (id) => {
    return api.get(`/menu/${id}`);
};


export const addMenuItem = (
    restaurantId,
    menuData
) => {
    return api.post(
        `/restaurants/${restaurantId}/menu`,
        menuData
    );
};


export const updateMenuItem = (
    id,
    menuData
) => {
    return api.put(
        `/menu/${id}`,
        menuData
    );
};


export const deleteMenuItem = (id) => {
    return api.delete(
        `/menu/${id}`
    );
};


export const searchMenu = (keyword) => {
    return api.get(
        `/menu/search?keyword=${encodeURIComponent(keyword)}`
    );
};